### ISSUE TITLE
Loadbalancer naming doesnt support implicit DNS

### ISSUE THREAD

**Issue by jim-bach** [kubernetes/cloud-provider-openstack#609]
**Is this a BUG REPORT or FEATURE REQUEST?:**
/kind feature

**What happened**:
For some clouds/use-cases the loadbalancer names can be implicitly given a DNS record (automatically because its a VIP), the current naming convention does not support that. Albeit better than that Google specific aXXXXXXXXXXXXXXX value that existed prior.

**What you expected to happen**:
Sanitize and conform to patterns respectful of DNS naming conventions
https://tools.ietf.org/html/rfc952, 

**How to reproduce it (as minimally and precisely as possible)**:
Create a DNS name using any loadbalancer name created e.g. kube_service_

Can be found here:
https://github.com/kubernetes/cloud-provider-openstack/blob/master/pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go#L542

Proposal: 
sanitize/validate size and make DNS appropriate (e.g. no underscores) < 63
fmt.Sprintf("kube_service_%s_%s_%s", clusterName, service.Namespace, service.Name)

Thoughts?

**Comment by lingxiankong** [comment]
@jim-bach I didn't see there is difference between your proposal and the current name formatting, missed something? Could you please give an example of a Kubernetes Service name and your preferred load balancer name?

**Comment by jim-bach** [comment]
>I didn't see there is difference between your proposal and the current name formatting, missed something?

I posted the current name formatting that is used `fmt.Sprintf("kube_service_%s_%s_%s", clusterName, service.Namespace, service.Name)` 

>sanitize/validate size and make DNS appropriate (e.g. no underscores) < 63
And I listed the issue, but I can try to clarify further.

1) nothing prevents it from being > 63 characters, if i put a long cluster name, service name, and/or namespace name in it will exceed this value easily. Especially with things like using prometheus operator helm chart. Thus, sanitizing/truncating the name could be useful here. Eventually it is truncated anyways https://github.com/kubernetes/cloud-provider-openstack/blob/master/pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go#L552.
253 is the max for the entire domain IIRC. This is just even the name, I would cut this at 63.

2) underscores are not valid, so at a minimum changing to kube-service-%s-%s-%s would already be an upgrade

3) Sanitize any non-DNS allowed characters, although less likely and not as important but would be useful anyways. 

As we further discuss the issue I'll try to make sure to add additional constraints that we'd want to meet
